src/processor.cpp

```diff
@@ -0,0 +1,373 @@
+{
+    struct process_data;
+    using process_id = string_id<process_data>;
+
+    struct process_data {
+
+        process_data() :
+            was_loaded(false) {};
+
+        process_id id;
+        std::string name;
+        std::vector<itype_id> components;
+        std::string output;
+        int fuel_intake;
+        int duration;
+        //TODO: Skill used
+
+        bool was_loaded;
+        void load(JsonObject &jo, const std::string &src);
+        void check() const;
+
+    };
+
+    process_id get_process_id( std::string id )
+    {
+        return process_id( id );
+    }
+
+    generic_factory<process_data> processes_data("process type", "id");
+
+}
+
+void process_data::load(JsonObject &jo, const std::string &)
+{
+    mandatory(jo, was_loaded, "name", name);
+    mandatory(jo, was_loaded, "components", components);
+    mandatory(jo, was_loaded, "output", output);
+    mandatory(jo, was_loaded, "fuel_intake", fuel_intake);
+    mandatory(jo, was_loaded, "duration", duration);
+}
+
+void process_data::check() const
+{
+    for (const itype_id &elem : components) {
+        if (!item::type_is_defined( elem )) {
+            debugmsg("Invalid components \"%s\" in \"%s\".", elem.c_str(), id.c_str());
+        }
+    }
+    if (!item::type_is_defined( output )) {
+        debugmsg("Invalid output \"%s\" in \"%s\".", output.c_str(), id.c_str());
+    }
+    if ( !(fuel_intake >= 0) ) {
+        debugmsg("Invalid fuel intake \"%d\" in \"%s\".", fuel_intake, id.c_str());
+    }
+    if (!(duration > 0)) {
+        debugmsg("Invalid duration \"%d\" in \"%s\".", duration, id.c_str());
+    }
+}
+
+void processes::load(JsonObject &jo, const std::string &src)
+{
+    processes_data.load(jo, src);
+}
+
+void processes::check()
+{
+    processes_data.check();
+}
+
+void processes::reset()
+{
+    processes_data.reset();
+}
+
+// Processor namespace
+
+namespace
+{
+struct processor_data;
+using processor_id = string_id<processor_data>;
+
+struct processor_data {
+
+    processor_data() :
+        was_loaded( false ) {};
+
+    processor_id id; //furniture
+    furn_str_id next_type;
+    itype_id fuel_type;
+
+
+    std::vector<process_id> processes;
+
+    std::string full_message;
+    std::string empty_message;
+    std::string start_message;
+    std::string finish_message;
+    std::string processing_message;
+    std::string fail_message;
+
+    bool was_loaded;
+
+    void load( JsonObject &jo, const std::string &src );
+    void check() const;
+
+};
+
+
+processor_id get_processor_id( const tripoint &pos )
+{
+    return processor_id( g->m.furn( pos ).id().str() );
+}
+
+generic_factory<processor_data> processors_data( "processor type", "id" );
+
+}
+
+void processor_data::load( JsonObject &jo, const std::string & )
+{
+    mandatory( jo, was_loaded, "next_type", next_type );
+    mandatory( jo, was_loaded, "fuel_type", fuel_type );
+    mandatory( jo, was_loaded, "processes", processes, string_id_reader<process_id>() );
```

Comment:
Try changing `process_id` to just `process`.
I think the `string_id_reader` wraps it in ID automatically, so here you are getting an ID of an ID rather than ID of the object.